pierwsza wersja ze tak powiem zrobiona :)

jtermi opened the pull request.

jtermi committed:
pierwsza wersja ze tak powiem zrobiona :)

marcinkunert reviewed (changes requested):
Film z code review: https://javastartbootcamp.slack.com/archives/C04JX1NL08G/p1679830030910139

jtermi committed:
poprawilem bledy:
- wylapywane sa tylko wyjatki dotyczace tworzenia i funkcjonowania komputera
- rozbudowalem konstruktor wyjatku o dodatkowy parametr by mozna bylo przekaywac jakiego podzespolu dotyczy wyjatek.
- utworzylem metody walidacji poszczegolnych parametrow przy tworzeniu podzespolow koputera.

slawekludwiczak commented on src/main/java/pl/javastart/task/Memory.java:
```diff
@@ -0,0 +1,80 @@
+    private static final int OVERCLOCKING_STEP = 100;
+
+    private int size;
+    private int baseFrequency;
+    private double workingTemperature;
+    private double maxSafeTemperature;
+
+    private double overClockedTemperature;
+    private double overClockedFrequency;
```
Nie do końca rozumiem to rozróżnienie na dwa taktowania i dwie temperatury. Procesor, czy pamięć ma temperaturę i taktowanie, które zmieniają się przy zwiększaniu, bądź obniżaniu taktowania

slawekludwiczak reviewed (changes requested):
Mógłbyś uniknąć duplikacji wydzielając dodatkową klasę pośrednią OverclockableComponent, nie ucinam za to punktów, ale pokazałem na filmie, jak możnaby to rozwiązać.
Stosujesz niespójne specyfikatory dostępu, raz jest private, raz protected, raz domyślny. Klasę Main najlepiej byłoby przenieść do osobnego pakietu i wtedy wszystkie klasy komponentów i Computer mógłbyś przenieść do pakietu typu "components" i tam stosować specyfikator domyślny, wskazując ograniczenie do niego. Jako publiczne mógłbyś zdefiniować tylko konstruktory, czy metodę overclock i printInfo, z których chcemy korzystać w metodzie main

code review https://javastartbootcamp.slack.com/archives/C04JX1NL08G/p1680251139560659
